### PR TITLE
[vernac] Cleanup on interface of Vernacentries

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

**Comment by ejgallego** [comment]
Very interestingly the test suite failure is due to a failure to dump a global, need to investigate if that was a pre-existing bug masked by the catch this PR removes.

**Comment by ejgallego** [comment]
So indeed the concrete point of failure is in bug_3923:
```coq
Module MkCertRuntimeTypes (B  : TRIVIAL) : CERTRUNTIMETYPES B.

Module FieldStore := MkStore B.

Definition cert_fieldstore := FieldStore.St.t.
Axiom empty_fieldstore : cert_fieldstore.

End MkCertRuntimeTypes.

Extraction MkCertRuntimeTypes. (* Was leading to an uncaught Not_found *)
```
Note the comment! This was solved by the dump function by ignoring the `Not_found`, is that the right thing to do?

The original bug report https://github.com/coq/coq/issues/3923

File: plugins/extraction/extract_env.ml [hunk 1]
```diff
@@ -644,8 +644,12 @@ let separate_extraction lr =
 (*s Simple extraction in the Coq toplevel. The vernacular command
     is \verb!Extraction! [qualid]. *)
 
+let dump_qualid r =
+  let gr = Smartlocate.global_with_alias r in
+  Dumpglob.add_glob gr
+
 let simple_extraction r =
-  Vernacentries.dump_global CAst.(make (Constrexpr.AN r));
+  dump_qualid r;
```
Comment (SkySkimmer) [inline plugins/extraction/extract_env.ml]: IMO we could just remove the dump, the sibling functions (full_extraction etc) don't bother for instance.
Or we could dump after locate_ref so we don't have to duplicate the locate call for dumping.
Or even dump *inside* locate_ref? oO What will science come up with next

Comment (ejgallego) [reply plugins/extraction/extract_env.ml]: Umm, that's for sure a naîve question, but why does the locate fail?

Comment (SkySkimmer) [reply plugins/extraction/extract_env.ml]: It looks for globrefs but we're calling extraction on a module.

Comment (ejgallego) [reply plugins/extraction/extract_env.ml]: Ok, let's remove the dump then.

**Comment by ejgallego** [comment]
PR ready IMVHO, needs assignee

**Review by SkySkimmer — APPROVED** [review]
I'd prefer to invert the commit order so that we don't re-introduce the bug only to immediately remove it.

**Comment by ejgallego** [comment]
> I'd prefer to invert the commit order so that we don't re-introduce the bug only to immediately remove it.

Done, thanks!

**Pull request merged**